js/ui/appActivation.js

```diff
@@ -139,8 +145,10 @@ const AppActivationContext = new Lang.Class({
     _maybeClearSplash: function() {
         // Clear the splash only when we've waited at least 700ms,
         // and when the app has transitioned to the running state...
-        if (this._appStateId == 0 && this._timeoutId == 0)
+        if (this._appStateId == 0 && this._timeoutId == 0) {
+            this._resetWindowsVisibility();
```

Comment:
Should probably be done inside `_clearSplash()` as that's called from elsewhere too?